### PR TITLE
Addressing multiple dashboard issues: deps loading once, field conversion, etc.

### PR DESCRIPTION
**Pull request by kvch**
## What does this PR do?

This PR addresses 3 minor issues in dashboard loading:
1. Everything should be loaded once
2. Some fields have to be strings that were objects before
3. Replacing index names in dashboards is no longer an NDJSON

## Why is it important?

This prevented setup from working properly.

## Checklist

- [x] My code follows the style guidelines of this project
~~- [ ] I have commented my code, particularly in hard-to-understand areas~~
~~- [ ] I have made corresponding changes to the documentation~~
~~- [ ] I have made corresponding change to the default configuration files~~
~~- [ ] I have added tests that prove my fix is effective or that my feature works~~
~~- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: libbeat/dashboards/modify_json.go [hunk 1]
```diff
@@ -175,54 +174,99 @@ func ReplaceIndexInDashboardObject(index string, content []byte) []byte {
...
+	}
+
+	if kibanaSavedObject, ok := attributes["kibanaSavedObjectMeta"].(map[string]interface{}); ok {
+		ba := ReplaceIndexInSavedObject(logger, index, kibanaSavedObject)
+		attributes["kibanaSavedObjectMeta"] = ba
+	}
+
+	if visState, ok := attributes["visState"].(string); ok {
+		nya := ReplaceIndexInVisState(logger, index, visState)
+		attributes["visState"] = nya
```
Comment (jsoriano) [inline libbeat/dashboards/modify_json.go:201]: Do `ba` and `nya` mean something? :slightly_smiling_face:

File: libbeat/dashboards/modify_json.go [hunk 2]
```diff
@@ -175,54 +174,99 @@ func ReplaceIndexInDashboardObject(index string, content []byte) []byte {
...
+		return content
+	}
+
+	return b
 }
 
-func replaceInNDJSON(logger *logp.Logger, index string, line []byte) []byte {
-	if len(bytes.TrimSpace(line)) == 0 {
-		return line
+func ConvertToStr(content []byte) []byte {
```
Comment (jsoriano) [inline libbeat/dashboards/modify_json.go:213]: Nit. This name is a bit misleading, it receives a content as `[]byte` and returns a `[]byte`. What do you think about calling this something as `EncodeJSONObjects`?

**Review by jsoriano — APPROVED** [review]
Tested locally and works for me. :+1:

**Review by jsoriano — APPROVED** [review]

File: libbeat/dashboards/modify_json.go (hunk 1 above)
Comment (kvch) [reply libbeat/dashboards/modify_json.go]: Independently no, but together "banya" means old, mean woman :D :D :D

**Review by jsoriano — APPROVED** [review]
:+1:

**Pull request merged**